src/transformers/models/bart/modeling_tf_bart.py

```diff
@@ -886,35 +884,33 @@ def call(
         # embed positions
         positions = self.embed_positions(input_shape, past_key_values_length)
 
-        if inputs_embeds is None:
-            inputs_embeds = self.embed_tokens(inputs["input_ids"])
-        else:
-            inputs_embeds = inputs["inputs_embeds"]
+        if inputs["inputs_embeds"] is None:
+            inputs["inputs_embeds"] = self.embed_tokens(inputs["input_ids"])
 
-        hidden_states = inputs_embeds * self.embed_scale
-
-        # [bsz, seq_len] -> [bsz, 1, tgt_seq_len, src_seq_len]
-        combined_attention_mask = None
-        if input_shape[-1] > 1:
-            combined_attention_mask = _make_causal_mask(input_shape, past_key_values_length=past_key_values_length)
```

Comment:
Could we maybe instead of all those functionality just do:

```python
        if input_shape[-1] > 1:
            combined_attention_mask = _make_causal_mask(input_shape, past_key_values_length=past_key_values_length)
        else:
            combined_attention_mask = tf.ones(input_shape, dtype=tf.int32)
```

Reply:
Yes, it is feasible. I put these condition because I don't know how really works the model so I tried to apply the same logic everywhere.

Let's test with what you propose :)

Reply:
Just noticed that my merged PR is probably conflicting with this one: https://github.com/huggingface/transformers/pull/9256

Reply:
and my above code snippet should probably be corrected to

```python
        if input_shape[-1] > 1:
            combined_attention_mask = _make_causal_mask(input_shape, past_key_values_length=past_key_values_length)
        else:
            combined_attention_mask = _expand(tf.ones((input_shape[0], input_shape[1] + past_key_values_length), dtype=tf.int32), tgt_len=input_shape[1])
```